AVRO-3275: Clarify specification on naming

opwvhk opened the pull request:
Mirror of PR apache/avro#1439

opwvhk committed:
AVRO-3275: Clarify specification on naming

martin-g reviewed (approved):
Thank you, @opwvhk !
I will merge it once the upstream PR is merged!

opwvhk committed:
VRO-3275: Further clarify specification on naming

Adds an example, and mentions for all occurrences of the 'namespace'
attribute that it is optional.

opwvhk committed:
AVRO-3275: Further clarify specification on naming

Undo erroneously added file

RyanSkraba commented on content/en/docs/1.11.0/Specification/_index.md:
```diff
@@ -172,11 +172,59 @@ The null namespace may not be used in a dot-separated sequence of names. So the
+      "name": "explicitNamespace",
+      "type": {
+        "type": "fixed",
+        "name": "simple",
```
```suggestion
        "name": "Simple",
```

RyanSkraba commented on content/en/docs/1.11.0/Specification/_index.md:
```diff
@@ -172,11 +172,59 @@ The null namespace may not be used in a dot-separated sequence of names. So the
+        "type": "fixed",
+        "name": "simple",
+        "namespace": "explicit",
+        "doc": "A simple name (attribute) and a namespace (attribute); the fullname is 'explicit.Simple' (this is a different type than of the 'inheritNull' field)",
```
```suggestion
        "doc": "A simple name (attribute) and a namespace (attribute); the fullname is 'explicit.Simple' (this is a different type than of the 'inheritNull' field).",
```

So nitpicky for consistency :D

opwvhk commented:
As the changes are on the Avro website, this PR is no longer needed.
